src/scanoss/agent/snippet_scan.c

```diff
@@ -0,0 +1,477 @@
+    fileMode = atol(PQgetvalue(result, i, 5));
+    asprintf(&realName,"%s",PQgetvalue(result, i, 8)); //8 fileName
+    for(int j=0;j<strlen(realName);j++){
+        if(( realName[j]>='A' && realName[j]<='Z' ) ||
+        ( realName[j]>='a' && realName[j]<='z' ) ||
+        ( realName[j]>='0' && realName[j]<='9' ) || realName[j]=='.'); 
+        else   realName[j]='_';
+        
+    }
+    //Nothing to be done on folders entries
+      if (parent != realParent && (fileMode == ((1<<28)|(1<<13)|(1<<9)))){
+    
+      } else {
+      // Ensure that it is a real file
+      //fileMode & ((1<<28)|(1<<13)|(1<<9)) == 0
+      if (fileMode != ((1<<28)|(1<<13)|(1<<9))) {
+        pFileFK = atoi(PQgetvalue(result, i, 4));
+        if (pFileFK != 0){
+          RestoreTempFile(tempFolder, pFileFK,parent,realName);
+     
+        }
+          
+      }
+    }
+    free(realName);
+  }
+  PQclear(result);
+  
+
+  return (0);
+}
+
+
+
+
+
+
+/***********************************************
+ Usage():
+ Command line options allow you to write the agent so it works
+ stand alone, in addition to working with the scheduler.
+ This simplifies code development and testing.
+ So if you have options, have a Usage().
+ Here are some suggested options (in addition to the program
+ specific options you may already have).
+ ***********************************************/
+void Usage(char *Name)
+{
+  printf("Usage: %s [file|folder]\n", Name);
+  printf("  -i   :: initialize the database, then exit.\n");
+  printf("  -v   :: verbose (-vv = more verbose)\n");
+  printf("  -c   :: Specify the directory for the system configuration.\n");
+  printf("  -C   <file path/folder path> :: run from command line.\n");
+  printf("  -V   :: print the version info, then exit.\n");
+} /* Usage() */
+
+struct node *head = NULL;
+struct node *current = NULL;
+/* Linked list to search for deprecated licenses namee*/
+
+// insert link at the first location
+void insertMapping(char *oldName, char *newName)
+{
+  struct node *linkItem = (struct node *)malloc(sizeof(struct node));
+int m1,m2;
+  m1=asprintf(&linkItem->oldName, "%s", oldName);
+  m2=asprintf(&linkItem->newName, "%s", newName);
+  if((m1==0)||(m2==0))
+       LOG_ERROR("Can not assign old to new names");
+  linkItem->next = head;
+
+  head = linkItem;
+}
+
+int isEmpty()
+{
+  return head == NULL;
+}
+/**
+ *  \brief Finds the new name of a license given its deprecated one
+*/
+struct node *find(char *data)
+{
+  struct node *current = head;
+  if (head == NULL)
+  {
+    return NULL;
+  }
+
+  while (strcmp(current->newName, data))
+  {
+    if (current->next == NULL)
+    {
+      return NULL;
+    }
+    else
+    {
+      current = current->next;
+    }
+  }
+  return current;
+}
+
+void loadDeprecated(){
+  insertMapping("GPL-2.0+", "GPL-2.0-or-later");
+  insertMapping("GPL-1.0+", "GPL-1.0-or-later");
+  insertMapping("StandardML-NJ", "SMLNJ");
+  insertMapping("wxWindows", "WXwindows");
```

Comment:
Mappings might not be required any more as we recently updated the licenses to new SPDX naming.

Reply:
Mapping removed!